### PR TITLE
firefox: build with pipewire patch to enable screenshare for wayland users

### PR DESCRIPTION
**Pull request by absrd**
Enables webrtc pipewire screensharing with pipewire, xdg-desktop-portal and https://github.com/emersion/xdg-desktop-portal-wlr

### CODE REVIEW

**Comment by ericonr** [comment]
Can you add `[ci skip]` to the commit body, please? CI won't be able to get through building Firefox.

File: srcpkgs/firefox/template [hunk 1]
```diff
@@ -18,14 +18,14 @@ lib32disabled=yes
...
- pixman-devel libevent-devel libnotify-devel libvpx-devel
+ pixman-devel libevent-devel libnotify-devel libvpx-devel pipewire-devel
  libXrender-devel libXcomposite-devel libSM-devel libXt-devel rust-std
  libXdamage-devel freetype-devel $(vopt_if alsa alsa-lib-devel)
  $(vopt_if dbus dbus-glib-devel) $(vopt_if pulseaudio pulseaudio-devel)
  $(vopt_if startup_notification startup-notification-devel)
  $(vopt_if xscreensaver libXScrnSaver-devel)
  $(vopt_if sndio sndio-devel) $(vopt_if jack jack-devel)"
-depends="nss>=3.47.1 desktop-file-utils hicolor-icon-theme"
+depends="nss>=3.47.1 desktop-file-utils hicolor-icon-theme libpipewire>=0.3.0"
```
Comment (ericonr) [inline srcpkgs/firefox/template:28]: Does the pipewire dependency not get detected automatically?

Comment (absrd) [reply srcpkgs/firefox/template:28]: It does. Removed

**Comment by absrd** [comment]
> Does Fedora move fast enough that renewing this patch for each Firefox release is a viable thing to do?

Depends on what you mean by viable ^^

The pipewire 0.2 edition's history is here https://src.fedoraproject.org/rpms/firefox/history/firefox-pipewire.patch?identifier=f29

So far the pipewire 0.3 edition hasn't changed again: https://src.fedoraproject.org/rpms/firefox/history/firefox-pipewire-0-3.patch?identifier=master

Otherwise, maybe add the link to https://src.fedoraproject.org/rpms/firefox/blob/master/f/firefox-pipewire-0-3.patch in the patch to easily diff should the build faild.

**Comment by absrd** [comment]
Added a link to Fedora's master branch for firefox with commit, so a diff can be should there be any changes.

**Comment by jnbr** [comment]
Is there any attempt to get this patch or another solution to this problem into upstream webrtc?

**Comment by absrd** [comment]
> Is there any attempt to get this patch or another solution to this problem into upstream webrtc?

https://bugzilla.mozilla.org/show_bug.cgi?id=1430775

Doesn't seem so, so each distro are making their patches.

**Comment by ericonr** [comment]
The patches seem to have been upstreamed into WebRTC already, I think the issue is that Firefox is having issues properly linking to the library (that's what I understood from the bug report and the linked threads). We might need to verify that it applies cleanly for cross compiled platforms as well.

**Comment by klardotsh** [comment]
What testing would still be needed for this? Is it enough to fire up some cross-compile jobs against this branch (perhaps building x86_64 and aarch64) and make sure they still produce seemingly-valid packages? (I could also test the aarch64 build if I were to revive my Snapdragon laptop...)

**Comment by jnbr** [comment]
I started a local build with this patch for all architectures we support.
I'm still no fan of pulling in a 800 lines patch.
This is more than all our current patches for firefox combined and it is already unpleasant to maintain.

**Comment by jnbr** [comment]
This patch breaks cross builds. If there is interest in getting it in, please fix it.

**Comment by st3r4g** [comment]
I think it works ootb since FF 84

**Comment by ericonr** [comment]
Does seem to be the case according to https://github.com/emersion/xdg-desktop-portal-wlr/wiki/Screencast-Compatibility

**Comment by ericonr** [comment]
Closing this, then.

**Pull request closed without merging**